src/terminal/terminal.rs

```diff
@@ -678,7 +770,33 @@ where
         Ok(remainder)
     }
 
+    /// Draw lines at the given vertical offset, assuming that the lines they are replacing on the
+    /// screen are cleared. The slice of cells must contain enough cells for the requested lines. A
+    /// slice of the unused cells are returned.
+    #[cfg(feature = "scrolling-regions")]
+    fn draw_lines_over_cleared<'a>(
+        &mut self,
+        y_offset: u16,
+        lines_to_draw: u16,
+        cells: &'a [Cell],
+    ) -> io::Result<&'a [Cell]> {
+        let width: usize = self.last_known_area.width.into();
+        let (to_draw, remainder) = cells.split_at(width * lines_to_draw as usize);
+        if lines_to_draw > 0 {
+            let area = Rect::new(0, y_offset, width as u16, y_offset + lines_to_draw);
+            let old = Buffer::empty(area);
+            let new = Buffer {
+                area,
+                content: to_draw.to_vec(),
+            };
+            self.backend.draw(old.diff(&new).into_iter())?;
+            self.backend.flush()?;
```

Comment:
Is the flush necessary? What's the impact of doing / not doing this?

Reply:
As I understand it, the output to the terminal uses buffered IO. You need to flush to get the changes to take affect on the terminal immediately.

I guess you could do the flush at the very end of the top-level function if you wanted to minimize flushes. But the code as it exists seems to not worry too much about optimizing (i.e. minimizing) the number of flushes.